src/tools/clippy/clippy_utils/src/hir_utils.rs

```diff
@@ -126,6 +127,20 @@ impl<'a, 'tcx> SpanlessEq<'a, 'tcx> {
     pub fn eq_path_segments(&mut self, left: &[PathSegment<'_>], right: &[PathSegment<'_>]) -> bool {
         self.inter_expr().eq_path_segments(left, right)
     }
+
+    pub fn eq_modifiers(&mut self, left: TraitBoundModifiers, right: TraitBoundModifiers) -> bool {
```

Comment:
Could be simplified by using `std::mem::discrimant` but otoh that wouldn't be as 'robust' or 'future-proof'.

```rs
std::mem::discriminant(left.constness, right.constness)
    && std::mem::discriminant(left.polarity, right.polarity) 
```

Reply:
Oh, good point!

Reply:
You know what, I should actually try testing the code rather than copying it...